sled-agent/src/backingfs.rs

```diff
@@ -0,0 +1,178 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Operations for dealing with persistent backing mounts for OS data
+
+// On Oxide hardware, the root filesystem is backed by a ramdisk and
+// non-persistent. However, there are several things within the root filesystem
+// which are useful to preserve across reboots, and these are backed persistent
+// datasets on the boot disk.
```

Comment:
> these are backed persistent datasets on the boot disk.

to be clear, that's done by this implementation?

Reply:
Yes, that's right.